crates/stackable-versioned-macros/src/lib.rs

```diff
@@ -65,101 +110,272 @@ mod consts;
 ///     version(name = "v1alpha1"),
 ///     options(allow_unsorted)
 /// )]
-/// struct Foo {}
+/// struct Foo {
+///     bar: usize,
+/// }
 /// ```
 ///
-/// ### Field Actions
+/// ## Item Actions
 ///
-/// This library currently supports three different field actions. Fields can
+/// This library currently supports three different item actions. Items can
 /// be added, renamed and deprecated. The macro ensures that these actions
 /// adhere to the following set of rules:
 ///
-/// - Fields cannot be added and deprecated in the same version.
-/// - Fields cannot be added and renamed in the same version.
-/// - Fields cannot be renamed and deprecated in the same version.
-/// - Fields added in version _a_, renamed _0...n_ times in versions
-///   b<sub>1</sub>, b<sub>2</sub>, ..., b<sub>n</sub> and deprecated in
-///   version _c_ must ensure _a < b<sub>1</sub>, b<sub>2</sub>, ...,
-///   b<sub>n</sub> < c_.
-/// - All field actions must use previously declared versions. Using versions
-///   not present at the container level will result in an error.
+/// 1. Items cannot be added and deprecated in the same version.
+/// 2. Items cannot be added and changed in the same version.
+/// 3. Items cannot be changed and deprecated in the same version.
+/// 4. Items added in version _a_, renamed _0...n_ times in versions
+///    b<sub>1</sub>, ..., b<sub>n</sub> and deprecated in
+///    version _c_ must ensure _a < b<sub>1</sub>, ..., b<sub>n</sub> < c_.
+/// 5. All item actions must use previously declared versions. Using versions
+///    not present at the container level will result in an error.
 ///
-/// For fields marked as deprecated, two additional rules apply:
+/// For items marked as deprecated, one additional rule applies:
 ///
-/// - Fields must start with the `deprecated_` prefix.
-/// - The deprecation note cannot be empty.
+/// - Fields must start with the `deprecated_` and variants with the
+///   `Deprecated` prefix. This is enforced because Kubernetes doesn't allow
+///   removing fields in CRDs entirely. Instead, they should be marked as
+///   deprecated. By convention this is done with the `deprecated` prefix.
```

Comment:
**thought (non-blocker)**: This should probably be a property of the `k8s` feature, so that developers not generating CRDs can still opt-in to k8s API versioning.